threedi_schema_edits/controlled_structures.py

```diff
@@ -0,0 +1,178 @@
+from sqlalchemy.exc import OperationalError
+
+from ThreeDiToolbox.utils.threedi_database import ThreediDatabase
+from ThreeDiToolbox.utils.user_messages import messagebar_message
+
+logger = logging.getLogger(__name__)
+
+TABLE_CONTROL = 'table_control'
+RULE_OPERATOR_BOTTOM_UP = '>'
+RULE_OPERATOR_TOP_DOWN = '<'
+V2_WEIR_VIEW_TABLE = "v2_weir_view"
+V2_WEIR_TABLE = 'v2_weir'
+MEASURE_VARIABLE_WATERLEVEL = 'waterlevel'
+ACTION_TYPE_SET_CREST_LEVEL = 'set_crest_level'
+
+
+class ControlledStructures(object):
+
+    def __init__(self, flavor):
+        self.flavor = flavor
+        self._schema = None  # will passed to get_uri()
+
+    def get_uri(self, **kwargs):
+        """
+        :returns an QgsDataSourceURI() instance
+
+        kwargs :
+            'host' --> network address (postgres) or
+                file path location (spatialite)
+            'port' --> port for the network address. Can
+                be omitted for spatialite
+            'user_name' --> database credential. Can
+                be omitted for spatialite
+            'password' --> database credential. Can
+                be omitted for spatialite
+            'schema' --> database schema name
+
+         """
+
+        self._uri = QgsDataSourceURI()
+        host = kwargs['host']
+        port = kwargs['port']
+        database = kwargs['database']
+        username = kwargs['username']
+        password = kwargs['password']
+        self._schema = kwargs['schema']
+        if self.flavor == 'spatialite':
+            self._uri.setDatabase(host)
+        elif self.flavor == 'postgres':
+            self._uri.setConnection(host, port, database, username, password)
+        return self._uri
+
+    def start_sqalchemy_engine(self, kwargs):
+        """
+        kwargs :
+            'host' --> network address (postgres) or
+                file path location (spatialite)
+            'port' --> port for the network address. Can
+                be omitted for spatialite
+            'user_name' --> database credential. Can
+                be omitted for spatialite
+            'password' --> database credential. Can
+                be omitted for spatialite
+            'schema' --> database schema name
+        """
+        self.threedi_db = ThreediDatabase(kwargs, db_type=self.flavor)
+        self.engine = self.threedi_db.engine
+
+    def create_action_table(self, list_of_values):
+        """
+        Create an action table (string) of a list of
+        measure and action values.
+
+        Args:
+            (list) list_of_values: A list of values, containing measuring
+                                   values and action values.
+
+        Returns:
+            (str) action_table: A string representing the action table to
+                                be saved.
+        """
+        action_table = '{0};{1}'.format(list_of_values[0], list_of_values[1])
+        return action_table
+
+    def select_all_attributes(self, table_name, attribute_name):
```

Comment:
De naam vind ik wat ongelukkig. ``select`` associeer ik in qgis echt met een (sub-) selectie van features. Ook "selecteer" je niet alle attributen maar een.